vue/src/components/JoistUpload.vue

```diff
@@ -70,7 +70,7 @@ export default class JoistUpload extends Vue {
   }
 
   private setFiles() {
-    this.files = [...this.fileInput.files!];
+    this.files = Array.from(this.fileInput.files || []);
```

Comment:
This is a case where we know more information than TypeScript: `this.fileInput` isn't just a generic `HTMLInputElement`, it always has a `.files` property. However, I agree this is a better way to keep TS happy.